src/hooks/utils.js

```diff
@@ -151,7 +162,11 @@ export function useEnhancedReducer(reducer, initialState, props) {
     [reducer],
   )
   const [state, dispatch] = useReducer(enhancedReducer, initialState)
-  const dispatchWithProps = action => dispatch({props, ...action})
+  const propsRef = useLatestRef(props)
+  const dispatchWithProps = useCallback(
+    action => dispatch({props: propsRef.current, ...action}),
+    [propsRef],
+  )
```

Comment:
This is all just to memoize the `dispatch` function. Behavior stays the same, but now you can pass `dispatch` to dependency arrays.